From c3827076d0373b63268db5c9dcfaa01061bb9c13 Mon Sep 17 00:00:00 2001
From: Hans Holmberg <hans.holmberg@intel.com>
Date: Tue, 3 Jun 2014 13:36:54 +0200
Subject: [PATCH 055/117] atomisp_gmin_platform: fix missing efi parameter
 panic

Fix a bug that manifests itself as a kernel panic
when an EFI parameter is not found.

Change-Id: Iffaceff8c7b72c56b30317945a45e3f7f5b4e983
Signed-off-by: Hans Holmberg <hans.holmberg@intel.com>
---
 arch/x86/platform/intel-mid/atomisp_gmin_platform.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/x86/platform/intel-mid/atomisp_gmin_platform.c b/arch/x86/platform/intel-mid/atomisp_gmin_platform.c
index 10dd8dc..146a8a4 100644
--- a/arch/x86/platform/intel-mid/atomisp_gmin_platform.c
+++ b/arch/x86/platform/intel-mid/atomisp_gmin_platform.c
@@ -236,9 +236,10 @@ int getvar_int(struct device *dev, const char *var, int def)
 	int ret;
 
 	ret = gmin_get_config_var(dev, var, val, &len);
-	val[len] = 0;
-	if (!ret)
+	if (!ret) {
+		val[len] = 0;
 		ret = kstrtol(val, 0, &result);
+	}
 
 	return ret ? def : result;
 }
-- 
1.9.3

